public/components/model/model_tags_panel/model_tags_panel.tsx

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useMemo, useRef, useState, useCallback, useEffect } from 'react';
+import {
+  EuiButton,
+  EuiDescribedFormGroup,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiForm,
+  EuiHorizontalRule,
+  EuiLink,
+  EuiPanel,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+  htmlIdGenerator,
+} from '@elastic/eui';
+import { useForm, useFieldArray } from 'react-hook-form';
+
+import { TagKeyFormData } from '../types';
+import { BottomFormActionBar } from '../bottom_form_action_bar';
+import { TagKey } from '../../common';
+import { useModelTagKeys } from '../../model_list/model_list.hooks';
+
+import { ModelTagKeyField } from './model_tag_key_field';
+import { ModelSavedTagKey } from './model_saved_tag_key';
+
+const MAX_TAG_NUM = 10;
+
+interface ModelTagsPanelProps {
+  tagKeys: TagKey[];
+  onTagKeysChange?: (tagKeys: TagKey[]) => void;
+}
+
+export const ModelTagsPanel = ({ tagKeys, onTagKeysChange }: ModelTagsPanelProps) => {
+  const formIdRef = useRef(htmlIdGenerator()());
+  const [isEditMode, setIsEditMode] = useState(false);
+  // TODO: change to real fetch all tags from backend in API integration phase
+  const [, allTagKeys] = useModelTagKeys();
+
+  const { control, resetField, handleSubmit, formState, trigger } = useForm<TagKeyFormData>({
+    mode: 'onChange',
+    defaultValues: {
+      tagKeys: [],
+    },
+  });
+  const unSavedChangeCount = Object.keys(formState.dirtyFields.tagKeys || {}).length;
```

Comment:
Isn't `formState.dirtyFields.tagKeys` an array? Looks strange to do `Object.keys(array)`.